src/ORM/DataList.php

```diff
@@ -1480,8 +1545,11 @@ public function eagerLoad(...$relationChains): static
                 // Keep track of what we've seen before so we don't accidentally add a level 1 relation
                 // (e.g. "Players") to the chains list when we already have it as part of a longer chain
                 // (e.g. "Players.Teams")
-                $list->eagerLoadAllRelations[$item] = $item;
+                $list->eagerLoadAllRelations[$item] ??= null;
             }
+            // Set the callback for this chain
+            $list->eagerLoadAllRelations[$relationChain] = $callback;
```

Comment:
Note that this *will* override a callback with null if someone adds the same chain twice, the second time without a callback. For example:

```php
$list = $list->eagerLoad(['MyRelation.AnotherRelation' => $someCallback]);
$list = $list->eagerLoad('MyRelation.AnotherRelation');
```

This allows complex branching logic, where some branches can remove the filters/sorting/etc that other branches might apply.